vendor/src/github.com/docker/notary/tuf/data/roles.go

```diff
@@ -97,6 +113,26 @@ func ValidRole(name string) bool {
 	return false
 }
 
+// IsDelegation checks if the role is a delegation or a root role
+func IsDelegation(role string) bool {
+	targetsBase := fmt.Sprintf("%s/", ValidRoles[CanonicalTargetsRole])
+
+	whitelistedChars, err := regexp.MatchString("^[-a-z0-9_/]+$", role)
```

Comment:
If `IsDelegation` is called more than infrequently, it might be a good idea to precompile the regexp into a global variable, i.e. `var delegationRegexp = regexp.MustCompile("^[-a-z0-9_/]+$")`. This also has the advantage that a syntax error in the regexp won't cause `IsDelegation` to always return false (the program would instead panic at startup).